From 444991b95ed25d58c3cd1646fa823620380b6ce6 Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Wed, 14 Mar 2018 16:44:13 +0800
Subject: [PATCH] ovs-lib: fix install_dir()

The command "install" is not available in OpenWrt by default

Signed-off-by: Yousong Zhou <yszhou4tech@gmail.com>
---
 utilities/ovs-lib.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/utilities/ovs-lib.in
+++ b/utilities/ovs-lib.in
@@ -159,7 +159,10 @@ install_dir () {
     [ "${OVS_USER##*:}" != "" ] && INSTALL_GROUP="${OVS_USER##*:}"
 
     if test ! -d "$DIR"; then
-        install -d -m "$INSTALL_MODE" -o "$INSTALL_USER" -g "$INSTALL_GROUP" "$DIR"
+        mkdir -p "$DIR"
+        chmod "$INSTALL_MODE" "$DIR"
+        chown "$INSTALL_USER" "$DIR"
+        chgrp "$INSTALL_GROUP" "$DIR"
         restorecon "$DIR" >/dev/null 2>&1
     fi
 }
